oasislmf/model_preparation/summaries.py

```diff
@@ -823,3 +841,32 @@ def write_exposure_summary(
         f.write(json.dumps(exposure_summary, ensure_ascii=False, indent=4))
 
     return fp
+
+
+def remove_duplicate_perils(exposure_df, loc_per_cov, peril_groups):
+    """
+    Remove grouped peril codes in the case that all component peril codes are also provided.
+
+    :param exposure_df: source exposure dataframe
+    :type exposure df: pandas.DataFrame
+
+    :param loc_per_cov: dataframe column name covering the peril codes for an exposure
+    :type loc_per_cov: str
+
+    :param peril_groups: peril group codes and associated perils
+    :type peril_groups: dict
+
+    :return: dataframe with peril group code removed if all component perils are provided
+    :rtype: pandas.DataFrame
+    """
+    unique_loc_per_cov = exposure_df[loc_per_cov].unique()
+    to_replace = {}
+    for exposure_perils_set in unique_loc_per_cov:
+        exposure_perils = exposure_perils_set.split(';')
+        for peril_group in peril_groups:
+            if peril_group in exposure_perils and \
+                    all(peril in exposure_perils for peril in peril_groups[peril_group]):
```

Comment:
Apparently, it would be more "pythonic" to break line as follows:
```
if peril_group in exposure_perils and all(
    peril in exposure_perils for peril in peril_groups[peril_group]
):
```
Very minor issue; I can change this before merging.